### PR TITLE
Implement Forms

### PR DESCRIPTION
**Pull request by lorenzoworx**
## In this pull request, I have implemented the following changes:
-  Create a method called `current_user` in ApplicationController to make current user data available to all controllers.
- Create forms to perform the following functions:
    - Create a Post on behalf of the `current_user`
    - Create a comment on behalf of the `current_user`
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -5,6 +5,29 @@ def index
   end
 
   def show
+    # @user = User.find_by(id: params[:user_id])
     @post = Post.where(author_id: params[:user_id], id: params[:id]).first
+    # @current_user = current_user
```
Comment (anagudelogu) [inline app/controllers/posts_controller.rb]: - [ ] Please don't forget to remove old pieces of commented code, they make your code base look ugly and make it less readable. 🤓 🔧

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class LikesController < ApplicationController
+  def create
+    post = Post.find(params[:post_id])
+    @like = post.like.new(author: current_user)
+
+    if @like.save
+      redirect_to user_post_path(current_user, post)
```
Comment (anagudelogu) [inline app/controllers/likes_controller.rb]: - [ ] You did a great job working on this milestone! 💯 I spotted the bug you were speaking about with the likes:

If you try to like any `Post` written by the `current_user` no error happens. However, if you try to like any other user `Post`, you get the following error:

image

Let's debug this:

First, if you look closely at your function, you're setting up the creation of a new `Like` on behalf of the currently logged-in user. According to your application logs, this part works fine:

image

However, the problem arises when it successfully saves the post. You're trying to redirect to the `user_post_path` but you're not using the `Post` author's id, instead, you're using the `current_user` id:

image

 For example, if you're on the route: `users/5/posts/28` and you are the user with id `1`. When you like this post, your function is going to redirect you to `users/1/posts/28` But `Post` with id `28` doesn't belong to user `1` and thus, the error appears. 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @lorenzoworx, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @lorenzoworx  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Forms are working as expected :gear: 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**